core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -1635,12 +1635,15 @@
   public static final String CARBON_SEARCH_MODE_ENABLE_DEFAULT = "false";
 
   /**
-   * Num of threads used in query executor when using search mode.
+   * Thread size of static ExecutorService in each Node when using search mode.
+   * Default value is -1, it means that Executors.newCachedThreadPool() will be used to
+   * maximize utilization. If thread numbers has to be limited, set it a positive Integer
+   * will call Executors.newFixedThreadPool(int nThreads) instead
    */
   @CarbonProperty
   public static final String CARBON_SEARCH_MODE_THREAD = "carbon.search.mode.thread";
```

Comment:
please rename it to `CARBON_SEARCH_MODE_SCAN_THREAD`, it means the number of thread used to do the scan. There are other threads in thread mode, like handling the request from master